Mathlib/CategoryTheory/Abelian/GrothendieckAxioms.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2023 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Isaac Hernando, Coleton Kotch, Adam Topaz
+-/
+
+import Mathlib.CategoryTheory.Limits.Filtered
+import Mathlib.CategoryTheory.Limits.Preserves.Finite
+
+/-!
+
+# Grothendieck Axioms
+
+This file defines some of the Grothendieck Axioms for abelian categories, and proves
+basic facts about them.
+
+## Definitions
+
+1. `AB4` -- an abelian category satisfies `AB4` provided that coproducts are exact.
+2. `AB5` -- an abelian category satisfies `AB5` provided that filtered colimits are exact.
+
+## Results
+
+- `AB5` implies `AB4`
+
+## Implementation Details
+
+For `AB4` and `AB5`, we only assume left exactness as right exactness is automatic.
+
+## Projects
+
+- Add additional axioms.
+
+-/
+
+namespace CategoryTheory
+
+open Limits Classical
+
+universe v v' u u'
+
+variable (C : Type u) [Category.{v} C]
+
+/--
+A category `C` which has coproducts is said to have `AB4` provided that
+coproducts are exact.
+-/
+class AB4 [HasCoproducts C] where
```

Comment:
Should this be `extends HasCoproducts C`?

Reply:
I'm always unsure. I guess it depends on whether instances will rather have a `HasCoproducts` instance that's already defined prior to a `AB4` instance?

Reply:
I'm also unsure, you should ask on zulip

Reply:
This seems to have been resolved on zulip, let's keep it like this!